controllers/etcd_custodian_controller.go

```diff
@@ -168,6 +176,92 @@ func (ec *EtcdCustodian) updateEtcdStatus(ctx context.Context, logger logr.Logge
 	})
 }
 
+func calculatePDBminAvailable(etcd *druidv1alpha1.Etcd) int {
+	// do not enable in single mode
+	if etcd.Spec.Replicas < 2 {
+		return 0
+	}
+
+	// cluster is in bootstrap
+	if etcd.Status.ClusterSize == nil {
+		return 0
+	}
+
+	allMembersReady := false
+	backupReady := false
+	for _, condition := range etcd.Status.Conditions {
+		if condition.Type == druidv1alpha1.ConditionTypeAllMembersReady &&
+			condition.Status == druidv1alpha1.ConditionStatus(druidv1alpha1.EtcdMemberStatusReady) {
+			allMembersReady = true
+			continue
+		}
+
+		if condition.Type == druidv1alpha1.ConditionTypeBackupReady &&
+			condition.Status == druidv1alpha1.ConditionTrue {
+			backupReady = true
+			continue
+		}
+	}
+
+	clusterSize := int(*etcd.Status.ClusterSize)
+	values := make([]int, int(math.Floor(float64(clusterSize)/float64(2))+1))
+
+	if !allMembersReady {
+		readyMembers := 0
+		for _, member := range etcd.Status.Members {
+			if member.Status == druidv1alpha1.EtcdMemberStatusReady {
+				readyMembers += 1
+			}
+		}
+		values = append(values, readyMembers)
+	}
+
+	if backupReady {
+		values = append(values, clusterSize)
+	}
+
+	// calculate max value
+	max := values[0]
+	for i := 1; i < len(values); i++ {
+		if values[i] > max {
+			max = values[i]
+		}
+	}
+
+	return max
+}
+
+func (ec *EtcdCustodian) updatePodDisruptionBudget(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, refMgr *EtcdDruidRefManager) error {
+	pdb := &policyv1.PodDisruptionBudget{}
+	err := ec.Get(ctx, types.NamespacedName{Name: etcd.Name, Namespace: etcd.Namespace}, pdb)
+	if errors.IsNotFound(err) {
+		logger.Info("PDB is not yet created")
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	logger.Info("Claiming pdb object")
+	claimedPdb, err := refMgr.ClaimPodDisruptionBudget(ctx, pdb)
+	if err != nil {
+		return err
+	}
+
+	// determine the maximum minAvailable value
+	minAvailable := calculatePDBminAvailable(etcd)
+	if claimedPdb.Spec.MinAvailable.IntValue() == minAvailable {
+		// do not update, nothing changed
+		return nil
+	}
+
+	// update fields
+	converted := intstr.FromInt(minAvailable)
+	claimedPdb.Spec.MinAvailable = &converted
+	return ec.Update(ctx, claimedPdb)
```

Comment:
Can you use `patch` instead? Please find an example [here](https://github.com/gardener/etcd-druid/blob/f1cd8a82acde5d4c26d52448424f453d5b689019/controllers/etcd_controller.go#L499-L514 (you can leave out the `RetryOnConflict` part).